tests/bluetooth/tester/src/btp_mcp.c

```diff
@@ -0,0 +1,1307 @@
+
+	btp_send_mcp_icon_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_read_next_track_obj_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC read next track obj ID cb (%d)", err);
+
+	btp_send_mcp_next_track_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_set_next_track_obj_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC set next track obj ID cb (%d)", err);
+
+	btp_send_mcp_next_track_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_read_parent_group_obj_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC read parent group obj ID cb (%d)", err);
+
+	btp_send_parent_group_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_read_current_group_obj_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC read current group obj ID cb (%d)", err);
+
+	btp_send_current_group_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_set_current_group_obj_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC read current group obj ID cb (%d)", err);
+
+	btp_send_current_group_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_read_playing_order_cb(struct bt_conn *conn, int err, uint8_t order)
+{
+	LOG_DBG("MCC read playing order cb (%d)", err);
+
+	btp_send_mcp_playing_order_ev(conn, err ? 1 : 0, order);
+}
+
+static void mcc_set_playing_order_cb(struct bt_conn *conn, int err, uint8_t order)
+{
+	LOG_DBG("MCC set playing order cb (%d)", err);
+
+	btp_send_mcp_playing_order_ev(conn, err ? 1 : 0, order);
+}
+
+static void mcc_read_playing_orders_supported_cb(struct bt_conn *conn, int err, uint16_t orders)
+{
+	LOG_DBG("MCC set playing order cb (%d)", err);
+
+	btp_send_mcp_playing_orders_supported_ev(conn, err ? 1 : 0, orders);
+}
+
+static void mcc_media_state_read_cb(struct bt_conn *conn, int err, uint8_t state)
+{
+	LOG_DBG("MCC media state read cb (%d)", err);
+
+	btp_send_mcp_media_state_ev(conn, err ? 1 : 0, state);
+}
+
+static void mcc_opcodes_supported_cb(struct bt_conn *conn, int err, uint32_t opcodes)
+{
+	LOG_DBG("MCC opcodes supported cb (%d)", err);
+
+	btp_send_mcp_opcodes_supported_ev(conn, err ? 1 : 0, opcodes);
+}
+
+static void mcc_content_control_id_cb(struct bt_conn *conn, int err, uint8_t ccid)
+{
+	LOG_DBG("MCC Content control ID cb (%d)", err);
+
+	btp_send_mcp_content_control_id_ev(conn, err ? 1 : 0, ccid);
+}
+
+static void mcc_segments_object_id_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC Segments Object ID cb (%d)", err);
+
+	btp_send_segments_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_current_track_obj_id_read_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC Segments Object ID read cb (%d)", err);
+
+	btp_send_current_track_obj_id_ev(conn, err ? 1 : 0, id);
+}
+
+static void mcc_current_track_obj_id_set_cb(struct bt_conn *conn, int err, uint64_t id)
+{
+	LOG_DBG("MCC Segments Object ID set cb (%d)", err);
+
+	btp_send_current_track_obj_id_ev(conn, err ? 1 : 0, id);
```

Comment:
`btp_send_current_track_obj_id_ev` and other claim the the `err` value is an ATT error code, but these are clearly always either `0` or `1` and not an ATT status.

Reply:
Right, I'll change them to 'status'

Reply:
What does the values 0 and 1 represent? BTP_STATUS_FAILED?

Reply:
Yes

Reply:
The please use those :)